packages/features/sales-escrows/pages/NFTPage/index.tsx

```diff
@@ -27,13 +27,74 @@ import {
   Banner,
   TabContent,
   ShowMoreBlock,
+  theme,
 } from '@origyn-sa/origyn-art-ui';
 import { useDialog } from '@connect2ic/react';
 import { getNftCollectionMeta, OrigynClient } from '@origyn-sa/mintjs';
 import { EscrowType } from '../../modals/StartEscrowModal';
 import { Principal } from '@dfinity/principal';
 import { PlaceholderImage } from '@dapp/common-assets';
 
+export interface CheckOfferProps {
+  odc: OdcDataWithSale;
+  onOpenEscrowModal: (escrowType: EscrowType) => void;
+}
+
+export const CheckOffer = ({ odc, onOpenEscrowModal }: CheckOfferProps) => {
+  const debug = useDebug();
+  const { principal, actor } = useContext(AuthContext);
+  const [existingOffer, setExistingOffer] = useState<any | null>(null);
+
+  const compareOfferSentWithSelectedToken = async () => {
+    const balance = await actor?.balance_of_nft_origyn({ principal });
```

Comment:
This returns a _BalanceResponse_ type if it's an ok response. Please check if the response has 'err' and handle appropriately, then strongly type the 'ok' value to _BalanceResponse_ and all the code that uses it.